source/Effect.cpp

```diff
@@ -58,6 +58,10 @@ void Effect::Load(const DataNode &node)
 			randomSpin = child.Value(1);
 		else if(child.Token(0) == "random frame rate" && child.Size() >= 2)
 			randomFrameRate = child.Value(1);
+		else if(child.Token(0) == "absolute angle" && child.Size() >= 2)
+			absoluteAngle = child.Value(1);
+		else if(child.Token(0) == "absolute velocity" && child.Size() >= 3)
+			absoluteVelocity = new Point(child.Value(1), child.Value(2));
```

Comment:
Don't use `new`! That's a memory leak waiting to happen. You can create a new data member, `hasAbsoluteVelocity` to check whether an absolute velocity was specified.